src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -45,6 +50,10 @@ public Connection(final String host) {
     this.host = host;
   }
 
+  public Connection(final File unixDomainSocket) {
```

Comment:
What if we take the `Socket` (and/or other objects) or a custom object which would create/connect a socket on demand? This would help to support many forms of sockets to be supported by Jedis at once.

Reply:
See last point in above comment.

Reply:
> you mention other sockets, any specific ones in mind?

Not anything specific, rather generic.

I just meant, if we had such constructor, sockets like unix domain socket, ssl or even plain socket with different configuration from Jedis could be used by users.

Reply:
> I just meant, if we had such constructor, sockets like unix domain socket, ssl or even plain socket with different configuration from Jedis could be used by users.

We can do that, but I think the question is should we expose that to the user as part of the library?

Jedis tries to do such constructs under the hood and expose other options (ssl, passwords, etc...) and sets some defaults for each of these, we would risk exposing this and making the library a bit harder (or a bit more confusing) to use.

It doesn't make much difference to my case by the way, so I am not opposing this, I just want to make sure you would be okay with that (minimal) risk.

Reply:
> we would risk exposing this and making the library a bit harder (or a bit more confusing) to use.

I'm not sure I follow you.

Reply:
As a user, I would much rather set a minimal parameters (e.g. port number, UDS path, credentials), than creating/setting/maintaining a socket (more code and might not be setup with the best configuration unless I am an advanced user).

Not the strongest argument, but this is my intuition. What do you think?

Reply:
- About dependency:
Most users are not enthusiastic about more dependencies. So I'm a bit reluctant on this.

- About _not easy_ option:
At least it is an option which would allow avid users to avoid long routes (like extending Jedis/Connection classes, building custom Jedis). Let's take `sendCommand` method (coming in 3.1.0) as an example. It would require some coding for the user but would allow them to use new commands in Redis (which are not in Jedis) more easily without waiting for Jedis to include those.

These are my viewpoints.

Reply:
Alright, sounds good, I will refactor this to allow creating it with a Socket object for advanced use-cases, will update this soon :)

Reply:
@mina-asham Please take a look at #2036. WDYT?